logic.ipynb

```diff
@@ -1438,55 +1438,520 @@
        "</html>\n"
       ],
       "text/plain": [
-       "<IPython.core.display.HTML object>"
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "psource(pl_resolution)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(True, False)"
+      ]
+     },
+     "execution_count": 38,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pl_resolution(wumpus_kb, ~P11), pl_resolution(wumpus_kb, P11)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(False, False)"
+      ]
+     },
+     "execution_count": 39,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pl_resolution(wumpus_kb, ~P22), pl_resolution(wumpus_kb, P22)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Forward and backward chaining\n",
+    "Introducing this section, we said we will look at two algorithms to check if a sentence is entailed by the `KB`, \n",
+    "but here's a third one. \n",
+    "The difference here is that our goal now is to determine if a knowledge base of definite clauses entails a single proposition symbol *q* - the query.\n",
+    "There is a catch however, the knowledge base can only contain **Horn clauses**.\n",
+    "<br>\n",
+    "#### Horn Clauses\n",
+    "Horn clauses can be defined as a *disjunction* of *literals* with **at most** one positive literal. \n",
+    "<br>\n",
+    "A horn clause with exactly one positive literal is called a *definite clause*.\n",
```

Comment:
I think 'horn' should be capitalized. There are other cases below where the word is not capitalized, you may need to take care of them too.